src/plugins/discover/public/application/main/services/load_saved_search.ts

```diff
@@ -130,6 +150,17 @@ function updateBySavedSearch(savedSearch: SavedSearch, deps: LoadSavedSearchDeps
   if (!isEqual(currentFilters, validFilters)) {
     services.filterManager.setFilters(validFilters);
   }
+
+  // Combine global filters (if defined) and app filters when loading a saved search.
+  // We do the same in `updateSavedSearch` function when user updates the saved search.
+  const globalFilters = services.filterManager.getGlobalFilters();
+  if (globalFilters?.length) {
+    savedSearch.searchSource.setField('filter', [
+      ...globalFilters,
+      ...services.filterManager.getAppFilters(),
+    ]);
+  }
```

Comment:
```suggestion
  savedSearch.searchSource.setField('filter', services.filterManager.getFilters());
```
I might be overlooking something here, but is this logic necessary, or would it work equally to just always set the `savedSearch.searchSource` filters to the entire filter list using `services.filterManager.getFilters()`?